activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java

```diff
@@ -1,55 +1,60 @@
 package org.activiti.cloud.services.rest.controllers;
 
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+import static org.springframework.http.HttpStatus.UNPROCESSABLE_ENTITY;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import javax.servlet.http.HttpServletResponse;
 import org.activiti.api.model.shared.model.ActivitiErrorMessage;
 import org.activiti.api.runtime.model.impl.ActivitiErrorMessageImpl;
 import org.activiti.api.runtime.shared.NotFoundException;
-import org.activiti.cloud.services.core.ActivitiForbiddenException;
+import org.activiti.api.runtime.shared.UnprocessableEntityException;
+import org.activiti.core.common.spring.security.policies.ActivitiForbiddenException;
 import org.activiti.engine.ActivitiObjectNotFoundException;
 import org.activiti.image.exception.ActivitiInterchangeInfoNotFoundException;
 import org.springframework.hateoas.Resource;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 
-import javax.servlet.http.HttpServletResponse;
-
 @RestControllerAdvice
 public class RuntimeBundleExceptionHandler {
 
-    @ExceptionHandler(ActivitiObjectNotFoundException.class)
-    @ResponseStatus(HttpStatus.NOT_FOUND)
-    public Resource<ActivitiErrorMessage>  handleAppException(ActivitiObjectNotFoundException ex, HttpServletResponse response) {
-        response.setContentType(MediaType.APPLICATION_JSON_VALUE);
-        return new Resource<>(new ActivitiErrorMessageImpl(HttpStatus.NOT_FOUND.value(), ex.getMessage()));
-    }
-
     @ExceptionHandler(ActivitiInterchangeInfoNotFoundException.class)
-    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @ResponseStatus(NO_CONTENT)
     public Resource<ActivitiErrorMessage> handleAppException(ActivitiInterchangeInfoNotFoundException ex, HttpServletResponse response) {
-        response.setContentType(MediaType.APPLICATION_JSON_VALUE);
-        return new Resource<>(new ActivitiErrorMessageImpl(HttpStatus.NOT_FOUND.value(), ex.getMessage()));
+        response.setContentType(APPLICATION_JSON_VALUE);
+        return new Resource<>(new ActivitiErrorMessageImpl(NO_CONTENT.value(), ex.getMessage()));
```

Comment:
it was not found before.. but is it no content an error code?

Reply:
No, the initial code was mapping the exception to no content (204), which as you highlighted, is not an error code. I didn't changed that, merely just in the response body I was returning the same status.

I do think this should map to some error status indeed, but that change needs a bit investigation

Reply:
ok, i would suggest to keep the investigation about this specific change on a different pr. And to keep this pr  only for the 422 introduction on that api only.

Reply:
Okay, reverted, created issue for that to be checked: https://github.com/Activiti/activiti-cloud-runtime-bundle-service/issues/504